### PR TITLE
Update EXT_texture_norm16 test with uploading img src

### PR DESCRIPTION
**Pull request by shrekshao**
Related to #3179

Update EXT_texture_norm16 test with uploading `HTMLImageElement` with `gl.texImage2D`. Together with some other refinements.

This is tested with a Chromium prototype implementation with support for decoding16-bit-per-channel png.

Also tests for `ImageBitmap` is added but not called at this moment. Other potentially support types are missing at this moment.("htmlcanvaselement", "htmlvideoelement" etc.)

### CODE REVIEW

File: sdk/tests/conformance2/extensions/ext-texture-norm16.html [hunk 1]
```diff
@@ -207,17 +305,22 @@
...
     } else {
       testPassed("No EXT_texture_norm16 support -- this is legal");
     }
   }
+  shouldBeTrue("successfullyParsed");
+  _addSpan('<br /><span class="pass">TEST COMPLETE</span>');
+  if (_jsTestPreVerboseLogging) {
+      _bufferedLogToConsole('TEST COMPLETE');
+  }
+  notifyFinishedToHarness();
```
Comment (kenrussell) [inline sdk/tests/conformance2/extensions/ext-texture-norm16.html:318]: This block should just be "finishTest()".

File: sdk/tests/conformance2/extensions/ext-texture-norm16.html [hunk 2]
```diff
@@ -49,49 +105,77 @@
...
-  wtu.checkCanvasRect(gl, 0, 0, 1, 1, generateFormatColor(format, expectedValue));
+  let readbackBuf = new Uint16Array(4);
+  wtu.checkCanvasRect(gl, 0, 0, 1, 1, generateFormatColor(format, expectedValue, 0xffff), undefined, 0, readbackBuf, gl.UNSIGNED_SHORT);
 }
 
-function testNorm16Render(interalFormat, format, type, tolerance) {
+function testSNorm16TextureImageDisallowed(texImageFunc, internalFormat, format, type) {
+  gl.bindTexture(gl.TEXTURE_2D, textures[0]);
+  texImageFunc(internalFormat, format, type);
+  wtu.glErrorShouldBe(gl, gl.INVALID_ENUM, "texture bindings succeed");
```
Comment (kenrussell) [inline sdk/tests/conformance2/extensions/ext-texture-norm16.html:168]: The error message should be "uploads from DOM elements to 16-bit SNORM textures are disallowed" or similar.

**Review by kenrussell — COMMENTED** [review]
Nice work getting these 16-bit-per-channel PNG images uploading losslessly to 16-bit-per-channel textures.

Overall this looks good. Let's figure out first in #3179 whether we need to add querying functions to the extension regarding the supported kinds of uploads.

**Comment by kenrussell** [comment]
Also please fix the lint error - there's trailing whitespace on one line.

**Comment by shrekshao** [comment]
Thanks for the review! Will add the querying functions if it gets approved for spec discussion.

**Comment by shrekshao** [comment]
Referencing #3141

**Comment by shrekshao** [comment]
Close this as discussed at WebGL WG on 2020-12-03. These tests would be added to that of EXT_texture_norm16_dom_source (potentially naming)

**Pull request closed without merging**